rts/lra/lra-filters/src/main/java/io/narayana/lra/filter/ClientLRAResponseFilter.java

```diff
@@ -45,7 +46,11 @@ public void filter(ClientRequestContext requestContext, ClientResponseContext re
          * (note we never popped the context in the request filter so we don't need to push outgoingLRA
          */
         if (incomingLRA != null) {
-            Current.push(new URL(incomingLRA.toString()));
+            try {
+                Current.push(new URI(incomingLRA.toString()));
+            } catch (URISyntaxException e) {
+                throw new IOException(e);
```

Comment:
I think it would be good to mention in the `IOException` message what's happening. Like giving the context "incomingLRA {id} fails to be pushed to current which means ..."

Reply:
+1

Reply:
Here is the new log message

Reply:
hm... I don't think is good to log WARN when there is the `IOException` which means an error.

I think it's a kind of strange to get two messages saying the same. I think when `IOException` is enhance with the information why it happens then it's what's needed.

Reply:
@mmusgrov my only concern regarding this PR is here about `IOException`. I think the exception itself should carry information what was the reason for it. And then I think duplication of the errors in the log is not necessary. If there is some rationale for having it "twice" then it should be error as it has impact for the further functionality works correctly.

Reply:
Tell me what you want and I'll add it.

Reply:
I don't want to get into a long debate here but if the end user is putting an invalid context on his response then it seems reasonable to log a warning. But if you can state the line of code you want me to add then I will include it.

Reply:
I expect to get the text which is part of the warning now `'%s': the JAX-RS '%s' filter for method '%s'#'%s' contained an invalid LRA context ('%s')`. It's how the WFLY does e.g.

* https://github.com/wildfly/wildfly/blob/16.0.0.Final/transactions/src/main/java/org/jboss/as/txn/integration/JBossContextXATerminator.java#L120
* https://github.com/wildfly/wildfly/blob/16.0.0.Final/transactions/src/main/java/org/jboss/as/txn/logging/TransactionLogger.java#L241

If you stands for need of having both - the warn message being logged and the IOException beight thrown - then the IOException message would be possibly shorter version like: `JAX-RS '%s' filter for method '%s'#'%s' contained an invalid LRA context ('%s'), can't be formated to URL`